src/compiler/checker.ts

```diff
@@ -15596,6 +15596,9 @@ namespace ts {
                 case SyntaxKind.InterfaceDeclaration:
                 case SyntaxKind.ModuleDeclaration:
                 case SyntaxKind.TypeAliasDeclaration:

```

Comment:
There's also a case for an `import =` declaration that reports this error in global module augmentations.